.github/workflows/build.yml

```diff
@@ -90,45 +110,46 @@ jobs:
         run: |
           FEATURES="${{ matrix.db }}"
           OPTS="--no-default-features"
+          FEATURE_SET="standard"
           export CC=clang
 
-          if [ "${{ matrix.name }}" = "fips" ]; then
-            FEATURES="${FEATURES},fips"
-          elif [ "${{ matrix.name }}" = "ossl3" ]; then
-            FEATURES="${FEATURES},standard"
-          elif [ "${{ matrix.name }}" = "release" ]; then
-            FEATURES="${FEATURES},standard"
-            OPTS="--release"
-          elif [ "${{ matrix.name }}" = "dynamic" ]; then
-            FEATURES="${FEATURES},standard,dynamic"
-            OPTS="--release"
-          elif [ "${{ matrix.name }}" = "i686" ]; then
-            FEATURES="${FEATURES},standard,dynamic"
-            OPTS="--target i686-unknown-linux-gnu"
+          if [ "${{ matrix.name }}" = "i686" ]; then
+            OPTS="${OPTS} --target i686-unknown-linux-gnu"
           elif [ "${{ matrix.name }}" = "minimal" ]; then
-            FEATURES="${FEATURES},dynamic"
+            FEATURE_SET="minimal"
           elif [ "${{ matrix.name }}" = "pqc" ]; then
-            FEATURES="${FEATURES},standard,pqc"
+            FEATURES="${FEATURES},pqc"
           elif [ "${{ matrix.name }}" = "no_sha1" ]; then
-            FEATURES="${FEATURES},standard,no_sha1"
+            FEATURES="${FEATURES},no_sha1"
+          fi
+
+          if [ "${{ matrix.linking }}" = "dynamic" ]; then
+            FEATURES="${FEATURES},${FEATURE_SET},dynamic"
+          elif [ "${{ matrix.linking }}" = "static" ]; then
+            FEATURES="${FEATURES},${FEATURE_SET}"
+          elif [ "${{ matrix.linking }}" = "fips" ]; then
+            FEATURES="${FEATURES},fips"
+          fi
+
+          if [ "${{ matrix.build }}" = "release" ]; then
+            OPTS="${OPTS} --release"
           fi
 
           # create a bugus configuration file to make sure it is not used during tests
           mkdir -p ~/.config/kryoptic/
           cp testdata/test.conf ~/.config/kryoptic/token.conf
 
           cargo build -vv $OPTS --features "$FEATURES"
-          KRYOPTIC_TRACE=trace.log cargo test -vv $OPTS --features "$FEATURES,log"
+          cargo test -vv $OPTS --features "$FEATURES"
```

Comment:
this removes testing of the `log` feature. I think we should add it to the matrix.